### PR TITLE
#160110741 Bug when adding an asset

### PR DESCRIPTION
**Pull request by HawiCaesar**
## What does this PR do?
- Fix a bug when adding an asset
## Description of Task to be completed?
- Break off adding an asset and adding an asset spec
## How should this be manually tested?
- Login to ART
- Click on `ADD ASSET` 
- Add relevant fields. 
- Click on `SAVE`. You should see a Success message when adding an asset.
## What are the relevant pivotal tracker stories?
[#160110741](https://www.pivotaltracker.com/story/show/160110741)
## Any background context you want to add?
n/a
## Important notes
n/a
## Packages installed
n/a
## Deployment note
n/a
## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [x] Test
- [ ] Documentation
## Screenshots

### CODE REVIEW

File: src/__test__/components/AddAssetComponent.test.js [hunk 1]
```diff
@@ -189,6 +189,32 @@ describe('<AddAssetContainer />', () => {
...
+    specsComponentSetup();
+    const returnSpecsState = () => wrapper.state().specs;
+    wrapper.setProps({
+      createAsset: () => returnSpecsState()
+    });
+
+    expect(returnSpecsState()).toEqual(wrapper.state().specs);
+
+    wrapper.find('DropdownComponent .item').at(3).simulate('click');
+    wrapper.find('input[type="radio"][name="processorType"]').at(0).simulate('change');
```
Comment (sachungo) [inline src/__test__/components/AddAssetComponent.test.js]: You can add a data attribute to the inputs named `processorType` and use that for testing. This way, you don't have to specifically get the radio input of the `processorType`. E.g.
In `My component`, I may have the following:
```
<label data-test="processor-type-input">
   <input type="radio" name="my-radio-button" checked />
   Testing 1
</label>
<label data-test="processor-type-input">
   <input type="radio" name="my-radio-button" />
   Testing 2
</label>
```

Then in `My test`, I'll do the following:
```
wrapper.find('[data-test="processor-type-input"]').first().simulate('change')
```

File: src/__test__/components/AddAssetComponent.test.js [hunk 2]
```diff
@@ -189,6 +189,32 @@ describe('<AddAssetContainer />', () => {
...
+    wrapper.find('input[type="radio"][name="processorSpeed"]').at(1).simulate('change');
+    wrapper.find('input[type="radio"][name="screenSize"]').at(1).simulate('change');
+    wrapper.find('input[type="radio"][name="storage"]').at(2).simulate('change');
+    wrapper.find('input[type="radio"][name="memory"]').at(2).simulate('change');
+    wrapper.find('Form').simulate('submit');
+    expect(wrapper.state().specs).toEqual({
```
Comment (sachungo) [inline src/__test__/components/AddAssetComponent.test.js]: Why not use `returnSpecsState` here?

**Comment by HawiCaesar** [comment]
@sachungo yes the changes fix the bug.

File: src/__test__/components/AddAssetComponent.test.js (hunk 2 above)
Comment (HawiCaesar) [reply src/__test__/components/AddAssetComponent.test.js]: I don't why I did not add this when committing. I will use `returnSpecsState`

**Pull request merged**